core/store/models/parse_randomness_request.go

```diff
@@ -0,0 +1,49 @@
+package models
+
+import (
+	"github.com/pkg/errors"
+
+	"chainlink/core/eth"
+	"chainlink/core/services/vrf"
+	"chainlink/core/utils"
+)
+
+// parseRandomnessRequest parses the RandomnessRequest log format.
+type parseRandomnessRequest struct{}
+
+var _ logRequestParser = parseRandomnessRequest{} // Implements logRequestParser
+
+// parseJSON returns the inputs to be passed as a JSON object to Random adapter
+func (parseRandomnessRequest) parseJSON(log eth.Log) (js JSON, err error) {
+	parsedLog, err := vrf.ParseRandomnessRequestLog(log)
+	if err != nil {
+		return JSON{}, errors.Wrapf(err,
+			"could not parse log data %#+v as RandomnessRequest log", log)
+	}
+	fullSeedString, err := utils.Uint256ToHex(parsedLog.Seed)
+	if err != nil {
+		return JSON{}, errors.Wrap(err, "vrf seed out of bounds")
+	}
+	return js.MultiAdd(KV{
+		// Address of log emitter
+		"address": log.Address.String(),
+		// Signature of callback function on consuming contract
+		"functionSelector": vrf.FulfillSelector(),
+		// Hash of the public key for the VRF to be used
+		"keyHash": parsedLog.KeyHash.Hex(),
+		// Raw input seed for the VRF (includes requester, nonce, etc.)
+		"seed": fullSeedString,
+		// The chainlink job corresponding to this VRF
+		"jobID": parsedLog.JobID.Hex(),
+		// Address of consuming contract which initially made the request
+		"sender": parsedLog.Sender.Hex(),
+	})
+}
+
+func (parseRandomnessRequest) parseRequestID(log eth.Log) string {
+	parsedLog, err := vrf.ParseRandomnessRequestLog(log)
+	if err != nil {
+		panic(errors.Wrapf(err, "while extracting randomness requestID from %#+v", log))
```

Comment:
Can we error instead of panicing?

Reply:
I can do that pretty easily, but two things to consider:

1. That would require changing the `logRequestParser` interface. 
2. This is called in `RunLogEvent#RunRequest`, after a call to `le.JSON` which results in a call to `parseJSON`, which already returns an error if `ParseRandomnessRequestLog` fails. So as the code is written, at least, this `panic` is purely defensive. There's an extra layer of defense, in that logs should only sent here via filtration on the `RandomnessRequestLogTopic`, and such logs should never fail `ParseRandomnessRequestLog`.

Reply:
I guess with these last two comments, there's a philosophical question of whether assertion failures are a good idea. I regard these as a form of testable documentation: Here are some properties of the current state which should be true, and which future execution depends on.